app-frontend/src/app/services/map/mapUtils.service.js

```diff
@@ -9,15 +14,13 @@ export default (app) => {
           */
         fitMapToProject(mapWrapper, project, offset = 0) {
             if (project.extent) {
-                let mapBounds = mapWrapper.map.getPixelBounds();
-                if (mapBounds.min.y !== mapBounds.max.y) {
+                mapWrapper.map.invalidateSize();
+                this.$timeout(() => {
                     mapWrapper.map.fitBounds(L.geoJSON(project.extent).getBounds(), {
-                        padding: [offset, offset]
+                        padding: [offset, offset],
+                        animate: false
                     });
-                } else {
-                    // eslint-disable-next-line
-                    setTimeout(this.fitMapToProject(mapWrapper, project, offset).bind(this), 125);
```

Comment:
It turns out that when we actually got into this recursion case, it would never resolve and thus would just result in a stack overflow.